source/guides/postgres-migration.rst

```diff
@@ -0,0 +1,208 @@
+
+Migrate the data
+----------------
+
+Once we set the schema to desired state, we can start migrating the **data** by running ``pgLoader`` \*\*
+
+\*\* Use the following configuration for the baseline of the data migration:
+
+.. code:: sql
+
+   LOAD DATABASE
+        FROM      mysql://{{ .mysql_user }}:{{ .mysql_password }}@mysql:3306/{{ .source_schema }}
+        INTO      pgsql://{{ .pg_user }}:{{ .pg_password }}@postgres:5432/{{ .target_schema }}
+
+   WITH data only,
+        workers = 8, concurrency = 1,
+        multiple readers per thread, rows per range = 50000,
+        create no tables,
+        create no indexes,
+        preserve index names
+
+   SET PostgreSQL PARAMETERS
+        maintenance_work_mem to '128MB',
+        work_mem to '12MB'
+
+   SET MySQL PARAMETERS
+         net_read_timeout  = '120',
+         net_write_timeout = '120'
+
+   CAST column Channels.Type to channel_type drop typemod,
+        column Teams.Type to team_type drop typemod,
+        column UploadSessions.Type to upload_session_type drop typemod,
+        column Drafts.Priority to text,
+        type int when (= precision 11) to integer drop typemod,
+        type bigint when (= precision 20) to bigint drop typemod,
+        type text to varchar drop typemod,
+        type tinyint when (<= precision 4) to boolean using tinyint-to-boolean,
+        type json to jsonb drop typemod
+
+   MATERIALIZE VIEWS exclude_products
+        excluding table names matching ~<IR_>, ~<focalboard>
+
+   BEFORE LOAD DO
+        $$ ALTER SCHEMA public RENAME TO {{ .source_schema }}; $$
+
+   AFTER LOAD DO
+        $$ UPDATE {{ .source_schema }}.db_migrations set name='add_createat_to_teamembers' where version=92; $$,
+        $$ ALTER SCHEMA {{ .source_schema }} RENAME TO public; $$;
+
+Once you save this configuration file, eg. ``migration.load``, you can run the ``pgLoader`` with following command:
+
+.. code:: bash
+
+   pgLoader migration.load > migration.log
+
+Feel free to contribute to and/or report your findings through your migration to us.
+
+Compare the data
+----------------
+
+We internally developed a tool to simplify the process of comparing contents of two databases. The ``dbcmp`` tool compares every table and reports whether if there is a diversity between two schemas.
+
+The tool includes a few flags to run a comparison:
+
+.. code:: sh
+
+   Usage:
+     dbcmp [flags]
+
+   Flags:
+         --exclude strings   exclude tables from comparison, takes comma-separated values.
+     -h, --help              help for dbcmp
+         --source string     source database dsn
+         --target string     target database dsn
+     -v, --version           version for dbcmp
+
+For our case we can simply run the following command:
+
+.. code:: sh
+
+   dbcmp --source "${MYSQL_DSN}" --target "${POSTGRES_DSN}" --exclude="db_migrations","ir_","focalboard","systems"
```

Comment:
comma-separated values would usually mean a single string with comma separated values. But here, it looks like separate string values separated by commas?

Reply:
Yeah I think that would be less confusing IMO.